kratos/tests/test_specifications_utilities.py

```diff
@@ -162,5 +251,35 @@ def test_specifications_utilities_conditions_dependencies(self):
         docu = KratosMultiphysics.Parameters("""{"SurfaceLoadCondition3D3N"   : "This is a pure displacement condition"}""")
         self.assertEqual(KratosMultiphysics.SpecificationsUtilities.GetDocumention(model_part).IsEquivalentTo(docu), True)
 
+    @KratosUnittest.skipUnless(dependencies_are_available,"StructuralMechanicsApplication is not available")
+    def test_specifications_utilities_conditions_dependencies_list(self):
+        current_model = KratosMultiphysics.Model()
+
+        model_part= current_model.CreateModelPart("Main")
+
+        node1 = model_part.CreateNewNode(1, 0.0,0.0,0.0)
+        node2 = model_part.CreateNewNode(2, 1.0,0.0,0.0)
+        node3 = model_part.CreateNewNode(3, 1.0,1.0,0.0)
+        prop1 = KratosMultiphysics.Properties(1)
+        model_part.AddProperties(prop1)
+        cond1 = model_part.CreateNewCondition("SurfaceLoadCondition3D3N", 1, [1,2,3], model_part.GetProperties()[1])
+        cond1.Initialize(model_part.ProcessInfo)
+
+        list_entities = KratosMultiphysics.Parameters("""{
+            "condition_list" :  ["SurfaceLoadCondition3D3N"]
+        }""")
+
+        self.assertEqual(model_part.GetNodalSolutionStepDataSize(), 0)
+        KratosMultiphysics.SpecificationsUtilities.AddMissingVariablesFromEntitiesList(model_part, list_entities)
+        self.assertEqual(model_part.GetNodalSolutionStepDataSize(), 3)
+
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_X), False)
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_Y), False)
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_Z), False)
+        KratosMultiphysics.SpecificationsUtilities.AddMissingDofsFromEntitiesList(model_part, list_entities)
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_X), True)
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_Y), True)
+        self.assertEqual(node1.HasDofFor(KratosMultiphysics.DISPLACEMENT_Z), True)
```

Comment:
Would it be possible to remove this test and other `Element` test, then use `DistanceCalculationElementSimplex` element from core so there won't be any dependency on the `StructuralMechanicsApplication`?

Reply:
We need a real element for testing

Reply:
`DistanceCalculationElementSimplex` this also has all the required stuff for this test ryt? (May be you have to implement the specification method in the element). Because there are lots of people who does not compile `StructuralMechanicsApplication`. :/

Reply:
> We need a real element for testing

This is a real element implementing an Eikonal equation. As an alternative we could use the distance convection element, which also belongs to the core and implements a pure convection formulation.

Reply:
Okay

Reply:
Note that in #9484 I'm adding the `GetSpecifications` to the `DistanceCalculationElementSimplex` for the purpose of testing. Maybe makes sense that we merge #9484 first and then we upgrade this test to get rid of the `StructuralMechanicsApplication` dependency.

Reply:
Okaydoki

Reply:
ping

Reply:
Yes, yes, you are right

Reply:
Now it is done